lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_localize_helper.rb

```diff
@@ -171,6 +172,89 @@ def self.verify_pr_diff(main, library, main_strings, lib_strings, source_diff)
+        #
+        def self.quick_lint(string_tag, lang)
+          if string_tag['formatted'] == 'false' && string_tag.content.include?('%%')
+            UI.important "Warning: [#{lang}] translation for '#{string_tag['name']}' has attribute formatted=false, but still contains escaped '%%' in translation."
+          end
+        end
+        private_class_method :quick_lint
+
+        # Download translations from GlotPress
+        #
+        # @param [String] res_dir The relative path to the `…/src/main/res` directory
+        # @param [String] glotpress_project_url The base URL to the glotpress project to download the strings from
+        # @param [Array<Hash{Symbol=>String}>] locales_map An array of locales to download. Each item in the array must be a
+        #                                      Hash with keys `:glotpress` and `:android` containing the respective locale codes.
+        #
+        def self.download_from_glotpress(res_dir:, glotpress_project_url:, locales_map:)
+          attributes_to_copy = %w[formatted] # Attributes that we want to replicate into translated `string.xml` files
+          orig_file = File.join(res_dir, 'values', 'strings.xml')
+          orig_xml = File.open(orig_file) { |f| Nokogiri::XML(f, nil, Encoding::UTF_8.to_s) }
+          orig_attributes = orig_xml.xpath('//string').map { |tag| [tag['name'], tag.attributes.select { |k, _| attributes_to_copy.include?(k) }] }.to_h
+
+          locales_map.each do |lang_codes|
+            UI.message "Downloading translations for '#{lang_codes[:android]}' from GlotPress (#{lang_codes[:glotpress]})..."
+            lang_dir = File.join(res_dir, "values-#{lang_codes[:android]}")
+            lang_file = File.join(lang_dir, 'strings.xml')
+            uri = URI.parse("#{glotpress_project_url.chomp('/')}/#{lang_codes[:glotpress]}/default/export-translations?filters[status]=current&format=android")
+            begin
+              # Download XML
+              xml = uri.open { |f| Nokogiri::XML(f.read.gsub("\t", '    '), nil, Encoding::UTF_8.to_s) }
+              # Process XML (text substitutions, replicate attributes, quick-lint string)
+              xml.xpath('//string').each do |string_tag|
+                apply_substitutions(string_tag)
+                orig_attributes[string_tag['name']]&.each { |k, v| string_tag[k] = v }
+                quick_lint(string_tag, lang_codes[:android])
+              end
+              xml.xpath('//string-array/item').each { |item_tag| apply_substitutions(item_tag) }
+
+              # Save
+              FileUtils.mkdir(lang_dir) unless Dir.exist?(lang_dir)
+              File.open(lang_file, 'w') { |f| xml.write_to(f, encoding: Encoding::UTF_8.to_s, indent: 4) }
+            rescue StandardError => e
+              UI.error "Error downloading #{lang_codes[:glotpress]} - #{e.message}"
+              FileUtils.rm_rf(File.join(res_dir, "values-#{lang_codes[:android]}"))
```

Comment:
If I read this correctly, we should be able to use this to DRY the code

```suggestion
              FileUtils.rm_rf(lang_dir)
```